core_test.go

```diff
@@ -809,6 +811,65 @@ func TestCore_Batch(t *testing.T) {
 	return
 }
 
+func TestCore_Backup(t *testing.T) {
+	var (
+		c   *Core
+		err error
+	)
+
+	if c, err = testInit(); err != nil {
+		t.Fatal(err)
+	}
+	defer testTeardown(c)
+
+	// Create 1k entries in the db so we have data to compare a backup against
+	for i := 0; i < 1000; i++ {
+		c.New(&testStruct{
+			UserID:    "user_" + strconv.Itoa(i),
+			ContactID: "contact_" + strconv.Itoa(i),
+
+			Foo: "foo_" + strconv.Itoa(i),
+			Bar: "bar_" + strconv.Itoa(i),
+		})
+	}
+
+	// Sleep for a few seconds to ensure we test created/update times copying as expected (we want to
+	// make sure that we aren't asigning new created/update at time values)
+	time.Sleep(2 * time.Second)
+
+	// Run the backup on the core db
+	if err = c.Backup(context.Background(), "data_backup", &testStruct{}); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll("data_backup")
+
+	var backup *Core
+	// Get the newly created backup db
+	if backup, err = New("test", "data_backup", &testStruct{}, "users", "contacts"); err != nil {
+		t.Fatal(err)
+	}
+
+	coreEntries := make(map[string]Value)
+	// Gather all the core entries for comparison
+	c.ForEach(func(key string, val Value) (err error) {
+		coreEntries[key] = val
+		return
+	})
+
+	backupEntries := make(map[string]Value)
+	// Gather all the backup entries for comparison
+	backup.ForEach(func(key string, val Value) (err error) {
+		backupEntries[key] = val
+		return
+	})
+
+	var equal bool
+	// Compare the core/backup entries and ensure that they are equal
+	if equal = reflect.DeepEqual(coreEntries, backupEntries); !equal {
```

Comment:
Just an ignorant question here: why not drop the `equal` variable?

```
if !reflect.DeepEqual(coreEntries, backupEntries) {
   // ... 
}
```

Reply:
Agreed, we could just check `if reflect.DeepEqual` since we don't use the return value for anything other than a conditional